### PR TITLE
#420 | Deeplink handling

### LINKED ISSUE #420: Handle deeplinks properly
Presently the application breaks when attempting to navigate to a deeplink when unauthenticated/logged out, e.g. https://wallet-staging.netlify.app/evm/send

This may be because the pinia store is not initialized at the time of the component loading. what we need is a solution to handle authorizxation which does not depend on pinia at all, and which can be called in a router guard (router > index.js). this solution should likely utilize localstorage (along with a new vue route meta `requiresAuthentication` field) to determine if a user is logged in, or perhaps we will need a method to communicate with wagmi/metamask.

also, user ID's are not currently removed from localstorage on logout, so the same account will get multiple  user id's in localstorage; this should be fixed. only one ID should exist per user.

the only routes which currently should be accessible to an unauth user are demo and the NFT detail page.

> [what we need is] to swap the Sign Out button for a Connect button, that could either:
> handle connection there (pop the wallet select modal)
> or
> send users to the connection page for them to connect in the normal flow

image

### PR DESCRIPTION
**Pull request by EJSG**
# Fixes #420 

## Description
This PR does a couple of things
1. users who are not logged in are now redirected to the homepage when attempting to access a route that requires auth
2. enable some pages (demo, nft details) to be viewed while not logged in
3. when viewing a page while not logged in, adds the ability to sign in using the left nav
4. slightly changes the logic for when to display the 'back' button in header to be simpler

## Test scenarios
- go to https://deploy-preview-421--wallet-staging.netlify.app/
    - make sure you are logged out / do not log in
- go to the following pages: [send](https://deploy-preview-421--wallet-staging.netlify.app/evm/send), [receive](https://deploy-preview-421--wallet-staging.netlify.app/evm/receive), [wallet](https://deploy-preview-421--wallet-staging.netlify.app/evm/wallet), [NFT inventory](https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-inventory)
    - you should be redirected home page
- go to https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-details?contract=0x1111111111111111111111111111111111111111&id=0987
    - you should be able to see the NFT details page
    - you should see 'Log in' on the nav menu rather than Log Out
- click wallet or collectibles in the nav
    - you should be directed home
- go back to https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-details?contract=0x1111111111111111111111111111111111111111&id=0987
- click log in in the nav
    - the login modal should pop
- log in using metamask
    -  you should see a success message
    - the nav should change to say 'log out'
    - you should be able to freely navigate around the site
- log out
- go back to https://deploy-preview-421--wallet-staging.netlify.app/evm/collectible-details?contract=0x1111111111111111111111111111111111111111&id=0987
- click log in
- log in using wallet connect
    -  you should see a success message
    - the nav should change to say 'log out'
    - you should be able to freely navigate around the site

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [x] I have tested for mobile functionality and responsiveness

### CODE REVIEW

File: src/components/evm/AppNav.vue [hunk 1]
```diff
@@ -206,7 +240,7 @@ export default defineComponent({
                     :src="require('src/assets/icon--nft.svg')"
                     :class="{
                         'c-app-nav__icon': true,
-                        'c-app-nav__icon--current-route': $route.name === 'evm-nfts',
+                        'c-app-nav__icon--current-route': $route.name === 'evm-nft-inventory',
```
Comment (EJSG) [inline src/components/evm/AppNav.vue:243]: fix nft icon in the nav not being changed when on the inventory page

File: src/pages/home/EVMLoginButtons.vue [hunk 2]
```diff
@@ -5,29 +5,18 @@ import { defineComponent } from 'vue';
 
 export default defineComponent({
     name: 'EVMLoginButtons',
-    setup(props, { emit }){
```
Comment (EJSG) [inline src/pages/home/EVMLoginButtons.vue:8]: set default chain has been moved to store setup file

File: src/router/index.js [hunk 3]
```diff
@@ -25,11 +25,19 @@ export default function (/* { store, ssrContext } */) {
...
             return { name: 'home' };
         }
+
+        if (to.meta.requiresAuth) {
+            const isAuthenticated = !!localStorage.getItem('account');
+
+            if (!isAuthenticated) {
+                return { name: 'home' };
+            }
+        }
```
Comment (Viterbo) [inline src/router/index.js:40]: perfect! That solves the problem of the back button

File: src/components/ExternalLink.vue [hunk 4]
```diff
@@ -11,8 +12,7 @@ const formattedText = computed(() => {
     const textIsAddress = /^0x[0-9a-fA-F]+$/.test(props.text);
 
     if (textIsAddress) {
-        const address = props.text;
-        return address.slice(0, 6) + '...' + address.slice(-4);
+        return getShortenedHash(props.text);
```
Comment (Viterbo) [inline src/components/ExternalLink.vue:15]: Nice!! much cleaner :+1:

**Review by Viterbo — APPROVED** [review]

**Review by pmjanus — APPROVED** [review]

**Pull request merged**